ostd/src/arch/x86/mod.rs

```diff
@@ -60,7 +60,12 @@ pub(crate) fn check_tdx_init() {
 }
 
 pub(crate) fn init_on_bsp() {
+    // SAFETY: this function is only called once on BSP.
+    unsafe {
+        crate::arch::trap::init(true);
+    }
     irq::init();
+    crate::trap::softirq::init();
```

Comment:
Is this a bugfix? Why does softirq is enabled on BSPs until this PR of using GS for cpu local storage?

Reply:
No, I just thought it's more reasonable to init softirq after irq.

Reply:
But why is `softirq::init` called in `init_on_bsp`? This means initializing softirq for every CPU.